code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+	desc = "The hippie's pizza of choice."
+	icon_state = "dankpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/doctor_delight = 5, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/dank
+
+/obj/item/food/pizza/dank/raw
+	name = "raw dank pizza"
+	icon_state = "dankpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/dank/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/dank, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/dank/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/dank, 6, 30)
+
+/obj/item/food/pizzaslice/dank
+	name = "dank pizza slice"
+	desc = "So good, man..."
+	icon_state = "dankpizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/sassysage
+	name = "sassysage pizza"
+	desc = "You can really smell the sassiness."
+	icon_state = "sassysagepizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 20, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 6)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/sassysage
+
+/obj/item/food/pizza/sassysage/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/sassysage, 6, 30)
+
+/obj/item/food/pizza/sassysage/raw
+	name = "raw sassysage pizza"
+	icon_state = "sassysagepizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | MEAT | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/sassysage/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/sassysage, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/sassysage
+	name = "sassysage pizza slice"
+	desc = "Deliciously sassy."
+	icon_state = "sassysagepizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/pineapple
+	name = "\improper Hawaiian pizza"
+	desc = "The pizza equivalent of Einstein's riddle."
+	icon_state = "pineapplepizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 20, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 6, /datum/reagent/consumable/pineapplejuice = 8)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapple" = 2, "ham" = 2)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | FRUIT | PINEAPPLE
+	slice_type = /obj/item/food/pizzaslice/pineapple
+
+/obj/item/food/pizza/pineapple/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.